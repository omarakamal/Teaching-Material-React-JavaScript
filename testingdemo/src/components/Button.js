import React from 'react'

function Button() {
    return (
        <div>
            <button data-testid="button">click menmj omar</button>
        </div>
    )
}

export default Button