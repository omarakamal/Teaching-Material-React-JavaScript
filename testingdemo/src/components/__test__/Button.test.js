import React from "react"
import ReactDOM from "react-dom"
import Button from "../Button"
import {render} from "@testing-library/react"
import renderer from "react-test-renderer"

it("renders without crashing",()=>{
    const div=document.createElement("div")
    ReactDOM.render(<Button></Button>,div)
})

it("renders button correctly",()=>{
    const{getByTestId}=render(<Button></Button>)
    expect(getByTestId("button")).toHaveTextContent("click me")
})


it("matches snapshot",()=>{
    const tree = renderer.create(<Button></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})

