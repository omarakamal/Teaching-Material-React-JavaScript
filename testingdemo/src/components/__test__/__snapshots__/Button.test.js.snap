// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    data-testid="button"
  >
    click menmj omar
  </button>
</div>
`;
